controllers/storagecluster/reconcile.go

```diff
@@ -886,3 +890,14 @@ func getUnsupportedClientsCount(r *StorageClusterReconciler, namespace string) (
 
 	return count, nil
 }
+
+// Set dynamic default values in the StorageCluster CR
+func setDynamicDefaults(sc *ocsv1.StorageCluster) {
+
+	// Set default mon count value in the StorageCluster CR
+	if arbiterEnabled(sc) {
+		sc.Spec.ManagedResources.CephCluster.MonCount = defaults.ArbiterModeMonCount
+	} else if sc.Spec.ManagedResources.CephCluster.MonCount == 0 {
+		sc.Spec.ManagedResources.CephCluster.MonCount = defaults.DefaultMonCount
+	}
+}
```

Comment:
Looking at the implementation again, we already have a function to determine MON count here:
https://github.com/red-hat-storage/ocs-operator/blob/b0f1699d841c248f2ff868ed31e04b83a4708203/controllers/storagecluster/cephcluster.go#L717-L726
This function should be enough for our usecases. We do not need this new function to set Mon count on StorageCluster CR which is user defined.

@Nikhil-Ladha Is the feature not working without this change?

Reply:
> This function should be enough for our usecases. We do not need this new function to set Mon count on StorageCluster CR which is user defined.

Oh yes, seems like we can use this function here as well. Thanks for pointing it out, I will update the code to use this function.

> @Nikhil-Ladha Is the feature not working without this change?

The issue is in the UI, we are seeing a blank field in the UI when the alert is raised.
Shared the screenshot in this [comment](https://github.com/red-hat-storage/ocs-operator/pull/2362#issuecomment-1877091199)

Reply:
We can't fix that by updating StorageCluster spec. They need to read it from CephCluster.

Reply:
The reason we need this fix in StorageCluster spec is because the CephCluster spec is updated with the values we have in the StorageCluster CR.
So, when the alert is raised for CephLowMonNumber, we update that value in the StorageCluster CR, and not in the CephCluster CR directly. That's the reason fetching the inital value from CephCluster CR and later updating the StorageCluster CR doesn't seem like the best option here.

Reply:
Reading from CephCluster & Updating in StorageCluster may not be convenient for UI but IMO reading it from the CephCluster gives the correct number reliably and will avoid this complicated change.

Reply:
@malayparida2000 that is not convenient in the UI as well.

Reply:
> The reason we need this fix in StorageCluster spec is because the CephCluster spec is updated with the values we have in the StorageCluster CR.
>
This is not always true. Mon count is determined internally unless it is explicitly defined in StorageCluster.
It is an optional field. For all optional fields in StorageCluster, operator decides the corresponding CephCluster value.

> So, when the alert is raised for CephLowMonNumber, we update that value in the StorageCluster CR, and not in the CephCluster CR directly. That's the reason fetching the inital value from CephCluster CR and later updating the StorageCluster CR doesn't seem like the best option here.
>
It may not be the best option, but it is the simplest option we have.
Operators should not edit their own CRs. So, determining Mon count internally and updating it on StorageCluster spec when required, is not an option here.

Alternatively, we can provide the current Mon count as `StorageCluster.Status.CurrentMonCount`. This means, when reconciling CephCluster CR, get the Mon count and write it to `StorageCluster.Status.CurrentMonCount`.

Reply:
Providing in the status would be defiently better than the operator editing it's own CR

Reply:
Ok, in that case let's set the value in the Status of the StorageCluster CR, and I think UI can pick the value from there?
cc @bipuladh this should be fine, right?

Reply:
Updated the code to set the value in the Status of StorageCluster CR, under the field `CurrentMonCount`.